dftimewolf/lib/exporters/scp_ex.py

```diff
@@ -48,25 +59,112 @@ def SetUp(self, # pylint: disable=arguments-differ
     self._paths = paths.split(",")
     self._user = user
 
+    if direction not in ['upload', 'download']:
+      self.ModuleError(
+        'Parameter direction must be one of {upload, download}',
+        critical=True)
+    self._upload = direction == 'upload'
+
     if check_ssh and not self._SSHAvailable():
       self.ModuleError("Unable to connect to host.", critical=True)
 
   def Process(self):
-    """Copies the list of paths to the destination on user@hostname"""
-    dest = self._destination
-    user = ""
+    """Copies the list of paths to or from the remote host."""
+    if not self._paths:
+      if self._upload:
+        # We're uploading local paths to the remote host.
+        fspaths = self.state.GetContainers(containers.FSPath)
+      else:
+        # We're downloading remote paths to the local host.
+        fspaths = self.state.GetContainers(containers.RemoteFSPath)
+      self._paths = [fspath.path for fspath in fspaths]
+
+    if not self._paths:
+      self.ModuleError('No paths specified to SCP module.', critical=True)
+
+    self._CreateDestinationDirectory(remote=self._upload)
+
+    cmd = ['scp']
+    if self._id_file:
+      cmd.extend(['-i', self._id_file])
+    if self._upload:
+      # scp /path1 /path2 user@host:/destination
+      cmd.extend(self._paths)
+      cmd.extend(self._PrefixRemotePaths([self._destination]))
+    else:
+      # scp user@host:/path1 user@host:/path2 /destination
+      cmd.extend(self._PrefixRemotePaths(self._paths))
+      cmd.extend([self._destination])
+
+    self.logger.debug('Executing SCP command: {0:s}'.format(' '.join(cmd)))
+    ret = subprocess.call(cmd)
+    if ret != 0:
+      self.ModuleError(
+          'Failed copying {0!s}'.format(self._paths), critical=True)
```

Comment:
why the `!s` why not `{0:s}...... format(', '.join(self._paths).....`

Reply:
I'm really OK with the output being 

```
Failed copying ['path1', 'path2'] since it makes it clear what is part of the path and what isn't
```

I could do `{0:s}` and pass in `str(self._paths)` if you feel strongly about it.

Reply:
do I feel strongly about it... no... not at all